doc/crypto/api/ops/signature.rst

```diff
@@ -502,41 +504,73 @@ EdDSA signature algorithms
         
     This signature algorithm can be used with both the message and message with context  signature functions.
 
-    This calculates the Ed25519ctx algorithm as specified in :RFC-title:`8032#5.1`, and requires an Edwards25519 curve key. The `psa_sign_message()` and `psa_verify_message()` functions use an empty context string when computing or verifying signatures. 
+    This calculates the Ed25519ctx algorithm as specified in :RFC-title:`8032#5.1`, and requires an Edwards25519 curve key. The `psa_sign_message()` and `psa_verify_message()` functions use an zero-length context when computing or verifying signatures. 
     
-    To use a non-empty context, use the signature functions that accept a context parameter, such as :code:`psa_sign_message_with_context()` and :code:`psa_verify_message_with_context()`
+    To use a non-zero-length context, use the signature functions that accept a context parameter, such as :code:`psa_sign_message_with_context()` and :code:`psa_verify_message_with_context()`
 
     .. admonition:: Implementation note
 
-       Even if you do supply an empty context, signatures created with Ed25519ctx are distinct from those created with PureEdDSA.
+       Even if you supply an zero-length context, signatures created with Ed25519ctx are distinct from those created with PureEdDSA.
 
     .. subsection:: Usage
 
         This is a message signing algorithm. To calculate a signature, use one of the following approaches:
 
         *   Call `psa_sign_message()` or `psa_sign_message_with_context()` with the message.
 
-        Verifying a signature is similar, using, for example, `psa_verify_message()` or `psa_verify_message_with_contex()` instead of the signature function.
+        Verifying a signature is similar, using, for example, `psa_verify_message()` or `psa_verify_message_with_context()` instead of the signature function.
 
     .. subsection:: Compatible key types
 
         | :code:`PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_TWISTED_EDWARDS)`
         | :code:`PSA_KEY_TYPE_ECC_PUBLIC_KEY(PSA_ECC_FAMILY_TWISTED_EDWARDS)` (signature verification only)
 
 
-.. macro:: PSA_ALG_ED25519PH
+.. macro:: PSA_ALG_ED448CTX
```

Comment:
We want to enable algorithm/key-size agility by using a single algorithm identifier for "PureEdDSA with non-zero context": `PSA_ALG_EDDSA_CTX`. This identifier requires that the implementation perform Ed25519ctx with a 255-bit key and Ed448 with a 448-bit key.

This definition should be folded into the `PSA_ALG_ED25519CTX` and renamed to be key-size agnostic. We can use the 0x06000A00 value for `PSA_ALG_EDDSA_CTX`.